src/azure-cli/azure/cli/command_modules/aro/_aad.py

```diff
@@ -56,6 +61,15 @@ def get_service_principal(self, app_id):
         return None
 
     def create_service_principal(self, app_id):
-        return self.client.service_principals.create(ServicePrincipalCreateParameters(
-            app_id=app_id,
-        ))
+        max_retries = 3
+        retries = 0
+        while True:
+            try:
+                return self.client.service_principals.create(
+                    ServicePrincipalCreateParameters(app_id=app_id))
+            except GraphErrorException as ex:
+                if retries >= max_retries:
+                    raise
+                retries += 1
+                logger.warning("%s; retry %d of %d", ex, retries, max_retries)
+                time.sleep(10)
```

Comment:
This is very go-esque as written, I'd suggest a refactor to

```suggestion
        max_retries = 3
        for i in range(max_retries):
            try:
                return self.client.service_principals.create(
                    ServicePrincipalCreateParameters(app_id=app_id))
            except GraphErrorException as ex:
                logger.warning("%s; retry %d of %d", ex, i, max_retries)
                time.sleep(10)
```

Reply:
(Note that the quick refactor doesn't include that final raise, it just gives up -- would need to follow up with an appropriate exception)

Reply:
These refactor should be done in our repository, let's not diff this even more.